controllers/cinderapi_controller.go

```diff
@@ -199,6 +199,39 @@ func (r *CinderAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *CinderAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Watch for changes to any CustomServiceConfigSecrets
+	svcSecretFn := func(o client.Object) []reconcile.Request {
+		var namespace string = o.GetNamespace()
+		var secretName string = o.GetName()
+		result := []reconcile.Request{}
+
+		// get all API CRs
+		apis := &cinderv1beta1.CinderAPIList{}
+		listOpts := []client.ListOption{
+			client.InNamespace(namespace),
+		}
+		if err := r.Client.List(context.Background(), apis, listOpts...); err != nil {
+			r.Log.Error(err, "Unable to retrieve API CRs %v")
+			return nil
+		}
+		for _, cr := range apis.Items {
```

Comment:
Shouldn't we also be checking for `instance.Spec.Secret` and `instance.Spec.TransportURLSecret`?
Or are you thinking of adding that in another PR?

Reply:
Maybe we should add a comment mentioning how the other secrets are watched by the cinder-operator.

Reply:
Ack.